pkg/reconciler/instances/connectivityproxy/action.go

```diff
@@ -45,50 +51,60 @@ func (a *CustomAction) Run(context *service.ActionContext) error {
 		return errors.Wrap(err, "Error while retrieving binding from BTP Operator")
 	}
 
-	if binding != nil {
-		context.Logger.Debug("Reading ServiceBinding Secret")
-		bindingSecret, err := a.Loader.FindSecret(context, binding)
+	// detect if connectivity-proxy reconciliation should not be skipped
+
+	if binding == nil && app == nil {
+		return nil
+	}
 
-		context.Logger.Debug("Service Binding Secret check")
+	// detect if connectivity-proxy is being deleted
```

Comment:
Please remove secret `connectivity-sm-operator-secrets-tls` when connectivity-proxy is deleted

Suggested change change `removeIstioSecrets` into `removeInstalledSecrets`

```
func (a *CommandActions) removeInstalledSecrets(context *service.ActionContext) error {

	_, err := context.KubeClient.DeleteResource(context.Context, "secret", "cc-certs", "istio-system")
	if err != nil {
		context.Logger.Error("Error during removal of cc-certs in istio-system")
		return errors.Wrap(err, "Error during removal of cc-certs in istio-system")
	}

	_, err = context.KubeClient.DeleteResource(context.Context, "secret", "cc-certs-cacert", "istio-system")
	if err != nil {
		context.Logger.Info("Error during removal of cc-certs-cacert in istio-system")
		return errors.Wrap(err, "Error during removal of cc-certs-cacert in istio-system")
	}

	_, err = context.KubeClient.DeleteResource(context.Context, "secret", "connectivity-sm-operator-secrets-tls", "kyma-system")
	if err != nil {
		context.Logger.Info("Error during removal of connectivity-sm-operator-secrets-tls in kyma-system")
		return errors.Wrap(err, "Error during removal of connectivity-sm-operator-secrets-tls in kyma-system")
	}
	return nil
}
```